controlpanel/api/aws.py

```diff
@@ -349,23 +351,20 @@ def grant_bucket_access(role_name, bucket_arn, access_level, path_arns=[]):
     role = boto3.resource('iam').Role(role_name)
     policy = S3AccessPolicy(role.Policy('s3-access'))
     policy.revoke_access(bucket_arn)
+    policy.grant_list_access(bucket_arn)
     for arn in path_arns:
-        policy.grant_access(arn, access_level)
+        policy.grant_object_access(arn, access_level)
     policy.put()
 
 
 def revoke_bucket_access(role_name, bucket_arn=None, path_arns=[]):
```

Comment:
I'm a bit confused by this method, was this causing problems? is the `path_arns` still relevant/used? a part from that in the `if` statement it doesn't seem like it's used.

Do the client code still pass this `path_arns` in and expect something to happen?

Reply:
Yes, when bucket access was revoked, access to individual paths was still retained. `path_arns` is no longer used with these changes so it can be removed.